test/torchaudio_unittest/datasets/vctk_test.py

```diff
@@ -0,0 +1,104 @@
+    'The rainbow is a division of white light into many beautiful colors',
+    'These take the shape of a long round arch, with its path high above, and its two ends \
+        apparently beyond the horizon',
+    'There is, according to legend, a boiling pot of gold at one end'
+]
+
+
+class TestVCTK(TempDirMixin, TorchaudioTestCase):
+    backend = 'default'
+
+    root_dir = None
+    samples = []
+
+    _ext_txt = ".txt"
+    _ext_audio = ".wav"
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        dataset_dir = os.path.join(cls.root_dir, vctk.FOLDER_IN_ARCHIVE)
+        os.makedirs(dataset_dir, exist_ok=True)
+        sample_rate = 96000  # 96 kHz
+        seed = 0
+
+        for speaker in range(225, 230):
+            speaker_id = 'p' + str(speaker)
+            audio_path = os.path.join(dataset_dir, cls._folder_audio, speaker_id)
+            os.makedirs(audio_path, exist_ok=True)
+
+            file_path = os.path.join(dataset_dir, cls._folder_txt, speaker_id)
+            os.makedirs(file_path, exist_ok=True)
+
+            for utterance_id in range(1, 11):
+                filename = f'{speaker_id}_{utterance_id:03d}_mic1'
+                audio_file_path = os.path.join(audio_path, filename + cls._ext_audio)
+
+                data = get_whitenoise(
+                    sample_rate=sample_rate,
+                    duration=0.01,
+                    n_channels=1,
+                    dtype='float32',
+                    seed=seed
+                )
+                save_wav(audio_file_path, data, sample_rate)
+
+                txt_file_path = os.path.join(file_path, filename[:-5] + cls._ext_txt)
+                utterance = UTTERANCE[utterance_id-1]
+                with open(txt_file_path, 'w') as f:
+                    f.write(utterance)
+
+                sample = (
+                    normalize_wav(data),
+                    sample_rate,
+                    utterance,
+                    speaker_id,
+                    utterance_id
+                )
+                cls.samples.append(sample)
+
+                seed += 1
+
+    @classmethod
+    def tearDownClass(cls):
+        # In case of test failure
+        vctk.VCTK._ext_audio = '.flac'
+
+    def test_vctk(self):
+        vctk.VCTK._ext_audio = '.wav'
+        dataset = vctk.VCTK(self.root_dir)
+        num_samples = 0
+        for i, (data, sample_rate, utterance, speaker_id, utterance_id) in enumerate(dataset):
+            self.assertEqual(data, self.samples[i][0], atol=5e-5, rtol=1e-8)
+            assert sample_rate == self.samples[i][1]
+            assert utterance == self.samples[i][2]
+            assert speaker_id == self.samples[i][3]
+            assert int(utterance_id) == self.samples[i][4]
+            num_samples += 1
+
+        assert num_samples == len(self.samples)
+        vctk.VCTK._ext_audio = '.flac'
```

Comment:
You do not need to do this in test method as it is done in `tearDownClass`.

Reply:
corrected